t/apicast.t

```diff
@@ -453,152 +401,119 @@ MHcCAQEEIFCV3VwLEFKz9+yTR5vzonmLPYO/fUvZiMVU1Hb11nN8oAoGCCqGSM49
-Including it's host so it is easy to see that configuration was loaded.
---- http_config
-  lua_package_path "$TEST_NGINX_LUA_PATH";
---- config
-location /t {
-  content_by_lua_block {
-    require('apicast.configuration_loader').global({
-      services = {
-        { id = 1, proxy = { hosts = { 'foo', 'bar' } } },
-        { id = 2, proxy = { hosts = { 'baz', 'daz' } } },
-      }
-    })
-    ngx.say('all ok')
-  }
+Including its host so it is easy to see that configuration was loaded.
+--- configuration
+{
+  "services" : [
+  { "id" : 1, "proxy" : { "hosts" : [ "foo", "bar" ] } },
+  { "id" : 2, "proxy" : { "hosts" : [ "baz", "daz" ] } }
+  ]
 }
 --- request
-GET /t
---- response_body
-all ok
+GET /
+--- error_code: 404
 --- log_level: info
---- error_code: 200
 --- error_log
 added service 1 configuration with hosts: foo, bar
 added service 2 configuration with hosts: baz, daz
+--- no_error_log
+[error]
+
 
 === TEST 15: return headers with debugging info
-When X-3scale-Debug header has value of the backend authentication.
---- http_config
-  include $TEST_NGINX_UPSTREAM_CONFIG;
-  lua_package_path "$TEST_NGINX_LUA_PATH";
-  init_by_lua_block {
-    require('apicast.configuration_loader').mock({
-      services = {
-        {
-          id = 42,
-          backend_version = 1,
-          backend_authentication_type = 'service_token',
-          backend_authentication_value = 'service-token',
-          proxy = {
-            api_backend = "http://127.0.0.1:$TEST_NGINX_SERVER_PORT/api/",
-            backend = {
-                endpoint = 'http://127.0.0.1:$TEST_NGINX_SERVER_PORT'
-            },
-            proxy_rules = {
-              { pattern = '/', http_method = 'GET', metric_system_name = 'hits', delta = 2 }
-            }
-          }
-       },
+When X-3scale-Debug header has value of the backend authentication
+--- configuration
+{
+  "services" : [
+    {
+      "id": 42,
+      "backend_version": 1,
+      "backend_authentication_type": "service_token",
+      "backend_authentication_value" : "service-token",
+      "proxy": {
+        "api_backend": "http://test:$TEST_NGINX_SERVER_PORT/api-backend/",
+        "proxy_rules": [
+            { "pattern" : "/", "http_method" : "GET", "metric_system_name" : "hits", "delta" : 2} 
+        ]
       }
-    })
+    }
+  ]
+}
+--- upstream
+  location / {
+     echo 'yay, api backend!';
   }
---- config
-  include $TEST_NGINX_APICAST_CONFIG;
-  include $TEST_NGINX_BACKEND_CONFIG;
-
-  location /api/ {
-    echo "all ok";
+--- backend
+  location /transactions/authrep.xml {
+    content_by_lua_block { ngx.exit(200) } 
   }
 --- request
-GET /t?user_key=val
+GET /?user_key=somekey
 --- more_headers
 X-3scale-Debug: service-token
 --- response_body
-all ok
+yay, api backend!
 --- error_code: 200
---- no_error_log
-[error]
 --- response_headers
 X-3scale-matched-rules: /
 X-3scale-usage: usage%5Bhits%5D=2
+--- no_error_log
+[error]
+
 
 === TEST 16: uses endpoint host as Host header
```

Comment:
this test is kind of duplicated from https://github.com/3scale/APIcast/pull/1348

If it is duplicated, makes sense to remove one of them?

Reply:
I removed this one because the other test (async report) covers this one too (and then sends an additional request) so I think it makes more to only keep that one.